src/routes/study/study.controller.ts

```diff
@@ -79,42 +81,36 @@ const createStudy = async (req: Request, res: Response) => {
   const NOT_FOUND = '데이터베이스에 일치하는 요청값이 없습니다';
 
   try {
-    const {
-      title,
-      studyAbout,
-      weekday,
-      frequency,
-      location,
-      capacity,
-      categoryCode,
-      dueDate,
-    } = req.body;
-
     const { id } = req.user as { id: string };
 
-    if (
-      !title ||
-      !studyAbout ||
-      !weekday ||
-      !frequency ||
-      !location ||
-      !capacity ||
-      !categoryCode ||
-      !dueDate
-    )
-      throw new Error(BAD_REQUEST);
+    if (!Object.keys(req.body).length) throw new Error(BAD_REQUEST);
+    const allowedFields = [
+      'title',
+      'studyAbout',
+      'weekday',
+      'frequency',
+      'location',
+      'capacity',
+      'categoryCode',
+      'dueDate',
+    ];
+    let count = 0;
+    Object.keys(req.body).forEach((key) => {
+      if (allowedFields.includes(key)) {
+        count += 1;
+      } else {
+        throw new Error(BAD_REQUEST);
+      }
+    });
+    if (count < 8) throw new Error(BAD_REQUEST);
```

Comment:
```suggestion
    const keys = Object.keys(req.body);
    if (
      keys.some((key) => !allowedFields.includes(key)) ||
      keys.length < allowedFields.length
    ) throw new Error(BAD_REQUEST);
```

처럼 하면 더 깔끔하게 표현할 수 있을 것 같습니다!